src/library_glfw.js

```diff
@@ -1124,6 +1124,7 @@ var LibraryGLFW = {
  * GLFW FUNCTIONS
  ******************************************************************************/
   glfwInit__deps: ['emscripten_get_device_pixel_ratio', 'malloc', 'free'],
+  glfwInit__proxy: 'sync',
```

Comment:
I don't think we need to proxy the entire initialization over to main thread.
Although it contains the canvas object, but proxyWorker has some handling for canvas object.
Proxying over may change the logic of this initialization code
https://github.com/emscripten-core/emscripten/blob/3cd9a380124bf17845f46819a895ec55a9e6ecf8/src/proxyWorker.js#L333

Reply:
It does add a bunch of event listeners to the window objects. Does that not require proxying?

Reply:
event listeners can be attached to the window object in workers.
On the proxyclient side, these events can be posted over to the worker to perform any operations if needed.
https://github.com/emscripten-core/emscripten/blob/3cd9a380124bf17845f46819a895ec55a9e6ecf8/src/proxyClient.js#L341C41-L341C41

Reply:
I am not really a web person, so I have no idea how to proxy just this part
```js
(function updatePixelRatio(){
      window.matchMedia("(resolution: " + window.devicePixelRatio + "dppx)")
      .addEventListener('change', updatePixelRatio, {once: true});
      GLFW.onWindowContentScaleChanged(_emscripten_get_device_pixel_ratio());
      })();
```

Reply:
I think it will be likely be simplest to simply proxy ever function here.. rather than try to do anything bespoke.

The simplest way to do this is problem via a helper function at the end of the file.  See `handleWebGLProxying` in `library_webgl.js` for how to do this.